samples/matter/light_bulb/src/zcl_callbacks.cpp

```diff
@@ -23,11 +23,17 @@ void MatterPostAttributeChangeCallback(const chip::app::ConcreteAttributePath &a
 	AttributeId attributeId = attributePath.mAttributeId;
 
 	if (clusterId == OnOff::Id && attributeId == OnOff::Attributes::OnOff::Id) {
-		ChipLogProgress(Zcl, "Cluster OnOff: attribute OnOff set to %" PRIu8, *value);
-		GetAppTask().PostEvent(AppEvent(*value ? AppEvent::On : AppEvent::Off, *value, true));
+		ChipLogProgress(Zcl, "Cluster OnOff: attribute OnOff set to %u", *value);
+		GetAppTask().GetLightingDevice().InitiateAction(*value ? PWMDevice::ON_ACTION : PWMDevice::OFF_ACTION,
+								AppEvent::kEventType_Lighting, value);
 	} else if (clusterId == LevelControl::Id && attributeId == LevelControl::Attributes::CurrentLevel::Id) {
-		ChipLogProgress(Zcl, "Cluster LevelControl: attribute CurrentLevel set to %" PRIu8, *value);
-		GetAppTask().PostEvent(AppEvent(AppEvent::Level, *value, true));
+		ChipLogProgress(Zcl, "Cluster LevelControl: attribute CurrentLevel set to %u", *value);
+		if (GetAppTask().GetLightingDevice().IsTurnedOn()) {
+			GetAppTask().GetLightingDevice().InitiateAction(PWMDevice::LEVEL_ACTION,
```

Comment:
Isn't it where `LightingEvent` is supposed to be used? It would improve thread-safety of the solution. Also the event is currently handled, but doesn't seem to be used, right?